test/io-tests/test_io.py

```diff
@@ -0,0 +1,348 @@
+
+            time.sleep(0.5)
+
+
+def test_app_settings(session):
+    """
+    App settings should not reset when the db pool times out.
+
+    See: https://github.com/PostgREST/postgrest/issues/1141
+
+    """
+    with run(CONFIGSDIR / "app-settings.config") as postgrest:
+        # Wait for the db pool to time out, set to 1s in config
+        time.sleep(2)
+
+        uri = "/rpc/get_guc_value?name=app.settings.external_api_secret"
+        response = session.get(postgrest.baseurl + uri)
+
+        assert response.status_code == 200
+        assert response.text == '"0123456789abcdef"'
+
+
+def test_app_settings_reload(session, tmp_path):
+    "App settings should be reloaded when PostgREST is sent SIGUSR2."
+    config = (CONFIGSDIR / "sigusr2-settings.config").read_text()
+    configfile = tmp_path / "test.config"
+    configfile.write_text(config)
+
+    with run(configfile) as postgrest:
+        url = f"{postgrest.baseurl}/rpc/get_guc_value?name=app.settings.name_var"
+
+        response = session.get(url)
+        assert response.status_code == 200
+        assert response.text == '"John"'
+
+        # change setting
+        configfile.write_text(config.replace("John", "Jane"))
+        # reload
+        postgrest.process.send_signal(signal.SIGUSR2)
+
+        response = session.get(url)
+        assert response.status_code == 200
+        assert response.text == '"Jane"'
+
+
+def test_jwt_secret_reload(session, tmp_path):
+    "JWT secret should be reloaded when PostgREST is sent SIGUSR2."
+    config = (CONFIGSDIR / "sigusr2-settings.config").read_text()
+    configfile = tmp_path / "test.config"
+    configfile.write_text(config)
+
+    headers = jwtauthheader({"role": "postgrest_test_author"}, SECRET)
+
+    with run(configfile) as postgrest:
+        url = f"{postgrest.baseurl}/authors_only"
+
+        response = session.get(url, headers=headers)
+        assert response.status_code == 401
+
+        # change setting
+        configfile.write_text(config.replace("invalid" * 5, SECRET))
+        # reload
+        postgrest.process.send_signal(signal.SIGUSR2)
+
+        response = session.get(url, headers=headers)
+        assert response.status_code == 200
+
+
+def test_db_schema_reload(session, tmp_path):
+    "DB schema should be reloaded when PostgREST is sent SIGUSR2."
+    config = (CONFIGSDIR / "sigusr2-settings.config").read_text()
+    configfile = tmp_path / "test.config"
+    configfile.write_text(config)
+
+    headers = {"Accept-Profile": "v1"}
+
+    with run(configfile) as postgrest:
+        url = f"{postgrest.baseurl}/parents"
+
+        response = session.get(url, headers=headers)
+        assert response.status_code == 404
+
+        # change setting
+        configfile.write_text(
+            config.replace('db-schemas = "test"', 'db-schemas = "test, v1"')
+        )
+        # reload
+        postgrest.process.send_signal(signal.SIGUSR2)
+        postgrest.process.send_signal(signal.SIGUSR1)
```

Comment:
maybe a comment explaining why we send `SIGUSR1` here as well (the test is about `SIGUSR2` after all?) could be added. and it should be separated from `# reload`.

`SIGUSR2` reloads the config file - `SIGUSR1` then reloads the schema cache to verify that the config reload actually happenend. So one is what we want to test, while the other is just boilerplate.